compiler/rustc_typeck/src/check/upvar.rs

```diff
@@ -1053,6 +1053,51 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
     }
 }
 
+/// Truncate the capture so that the place being borrowed is in accordance with RFC 1240,
+/// which states that it's unsafe to take a reference into a struct marked `repr(packed)`.
+fn restrict_repr_packed_field_ref_capture<'tcx>(
+    tcx: TyCtxt<'tcx>,
+    param_env: ty::ParamEnv<'tcx>,
+    place: &Place<'tcx>,
+) -> Place<'tcx> {
+    let pos = place.projections.iter().enumerate().position(|(i, p)| {
+        let ty = place.ty_before_projection(i);
+
+        match p.kind {
```

Comment:
There are a lot of matches and ifs here. It'd be nice to have a comment indicating what you are looking for. Something like:

// Return true for fields of packed structs, unless those fields have alignment 1.